### PR TITLE
Add region masks for MALI Antarctic mesh

### PR DESCRIPTION
**Pull request by hollyhan**
Previously, the AIS mesh-generation test case didn't include a method that creates region masks. This PR includes the method to create region masks based on the Antarctica IMBIE1 and IMBIE2-ISMIP6 basins stored in `geometric_features` repository (https://github.com/MPAS-Dev/geometric_features/tree/master/geometric_data/landice/region). 

This PR goes in parallel with the PR that converts the rasterized format of the IMBIE2-ISMIP6 source basin to the vector format (https://github.com/MPAS-Dev/geometric_features/pull/180). 

This PR also includes changes in the user's guide and developer's guide, which were forgotten to be included in the previous PR (https://github.com/MPAS-Dev/compass/pull/324).

Note: There is also a minor change included in this PR, which is adding the `mesh_filename` attribute that allows users to define the mesh name only once at the top of the script. This change is unrelated to the region mask task, but @xylar and I decided to still include the change.

### CODE REVIEW

File: conda/compass_env/spec-file.template [hunk 1]
```diff
@@ -8,7 +8,7 @@ cartopy_offlinedata
 cmocean
 esmf=*={{ mpi_prefix }}_*
 ffmpeg
-geometric_features=0.5.0
+geometric_features=0.6.0
```
Comment (xylar) [inline conda/compass_env/spec-file.template:11]: @hollyhan could you mention this version change explicitly at the bottom of he description of the PR?

Comment (xylar) [reply conda/compass_env/spec-file.template:11]: Thanks!

**Comment by xylar** [comment]
@hollyhan, I should be able to test this in about an hour.

File: conda/compass_env/spec-file.template (hunk 1 above)
Comment (hollyhan) [reply conda/compass_env/spec-file.template:11]: Of course. Thanks for the reminder to update the PR description! I've also just added the links to the other two PRs that go parallel to this one.

**Comment by hollyhan** [comment]
> @hollyhan, I should be able to test this in about an hour.

Sounds great. Thanks!

File: compass/landice/tests/antarctica/mesh.py [hunk 2]
```diff
@@ -123,34 +123,46 @@ def run(self):
...
+                mesh_filename, '-p', 'ais-bedmap2']
         check_call(args, logger=logger)
 
         logger.info('creating graph.info')
-        make_graph_file(mesh_filename='Antarctica.nc',
+        make_graph_file(mesh_filename=mesh_filename,
                         graph_filename='graph.info')
 
+        # create a region mask
+        mask_filename = 'imbie_regionMasks.nc'
```
Comment (xylar) [inline compass/landice/tests/antarctica/mesh.py:156]: Could you add this and `ismip6_regionMasks.nc` as output files of the step?

File: docs/developers_guide/landice/api.rst [hunk 3]
```diff
@@ -11,6 +11,22 @@ landice
 Test Groups
 ^^^^^^^^^^^
 
+antarctica
+~~~~~~~~~~~~~~~~~~~~~~
```
Comment (xylar) [inline docs/developers_guide/landice/api.rst:15]: Please make the underline here a bit shorter to match the header.

File: docs/users_guide/landice/test_groups/antarctica.rst [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    # mesh density functions
+    use_speed = True
+    use_dist_to_grounding_line = True
+    use_dist_to_edge = False
+
+mesh_gen
+--------
+
+``landice/antarctica/mesh_gen`` creates a 8-80km variable resolution mesh.
+There is no model integration step.
```
Comment (xylar) [inline docs/users_guide/landice/test_groups/antarctica.rst:56]: Could you add something about what the name of the output file is (`Antarctica.nc`) and also something about the fact that IMBIE1 and ISMIP6 (similar to IMBIE2) region masks are generated, along with references or links to where these are defined?

File: docs/users_guide/landice/test_groups/antarctica.rst [hunk 5]
```diff
@@ -0,0 +1,56 @@
+.. _landice_antarctica:
+
+antarctica
+==========
+
+The ``landice/antarctica`` test group includes a test case for creating a
+mesh for Antarctica.
+
+.. figure:: images/antarctica_8to80km.png
```
Comment (xylar) [inline docs/users_guide/landice/test_groups/antarctica.rst:9]:
```suggestion
.. figure:: images/antarctica_1to80km.png
```
Or you need to rename the image file itself.

File: docs/developers_guide/landice/api.rst [hunk 6]
```diff
@@ -11,6 +11,22 @@ landice
...
+
+.. currentmodule:: compass.landice.tests.antarctica
+
+.. autosummary::
+   :toctree: generated/
+
+   Antarctica
+
+   mesh_gen.MeshGen
+   mesh_gen.MeshGen.run
```
Comment (xylar) [inline docs/developers_guide/landice/api.rst:25]: This should not be included since you didn't override this method.

**Review by xylar — CHANGES_REQUESTED** [review]
@hollyhan, wonderful work!  Just requesting a few small changes that don't affect the functionality.

I was able to run the test case on the first try (once I defined the correct place for my local landice database -- user error!).  I used `paraview_vtk_field_extractor.py` to take a look at the region masks:
```
paraview_vtk_field_extractor.py -f imbie_regionMasks.nc -m Antarctica.nc -v allOnCells --ignore_time -o vtk_imbie
paraview_vtk_field_extractor.py -f ismip6_regionMasks.nc -m Antarctica.nc -v allOnCells --ignore_time -o vtk_ismip6
```
They look exactly like what I expected.

**Comment by xylar** [comment]
@hollyhan, can we rename the PR to "Add region masks and documentation to MALI Antarctica mesh,", or something similar?

File: compass/landice/tests/antarctica/mesh.py (hunk 2 above)
Comment (matthewhoffman) [inline compass/landice/tests/antarctica/mesh.py:156]: How about making this something like `f'{mesh_filename}_imbie_regionMasks.nc'?  (And similar for ismip6 mask filename)

**Review by matthewhoffman — APPROVED** [review]
@hollyhan , this is a great addition to have region masks automatically being generated in COMPASS for MALI meshes for the first time.  This will greatly facilitate the process in the future so we can start using regionMasks more regularly.  I just have one minor suggestion.  I'm going to approve now, with the understanding that Xylar has already given a thorough review addressing details of the implementation that I wouldn't have caught and that he'll give it a final look over before it gets merged.

**Comment by xylar** [comment]
@hollyhan, is it time to rebase this an revive our conversation?  I kind of lost track of this one but I think it is needed before #397

**Comment by hollyhan** [comment]
Hi @xylar, 

Sorry I lost track of this long-overdue PR. I've addressed all the comments.

**Comment by xylar** [comment]
@hollyhan, I think you missed this comment:
https://github.com/MPAS-Dev/compass/pull/406#issuecomment-1150219794
> can we rename the PR to "Add region masks and documentation to MALI Antarctica mesh,", or something similar?

Also, the PR needs to be rebased onto the current `main` branch both to address some conflicts and to bring in linting and testing that is now required.

**Comment by xylar** [comment]
Also, compass has been updated to `geometric_featues 1.0.1` in the meantime so that will probably be what is involved in resolving the conflicts.  And the description of the PR needs to be changed to reflect that these changes make use of the features in the 2 listed `geometric_features` PRs but to no longer state that this PR brings in a new `geometric_features` version.

**Comment by hollyhan** [comment]
HI @xylar and @matthewhoffman, 

This PR has gone through revision/updates and is ready to be reviewed again. 

Below is the resulting messages of the test:
```
(compass_region_masks) pn2034311:mesh_gen hollyhan$ compass run
landice/antarctica/mesh_gen
compass calling: compass.landice.tests.antarctica.mesh_gen.MeshGen.run()
  inherited from: compass.testcase.TestCase.run()
  in /Users/hollyhan/Desktop/RESEARCH/MALI/compass/add_regionMask_AIS_meshGen/compass/testcase.py

compass calling: compass.run.serial._run_test()
  in /Users/hollyhan/Desktop/RESEARCH/MALI/compass/add_regionMask_AIS_meshGen/compass/run/serial.py

Running steps: mesh
  * step: mesh

compass calling: compass.landice.tests.antarctica.mesh_gen.MeshGen.validate()
  inherited from: compass.testcase.TestCase.validate()
  in /Users/hollyhan/Desktop/RESEARCH/MALI/compass/add_regionMask_AIS_meshGen/compass/testcase.py

  test execution:      SUCCESS
  test runtime:        11:25
Test Runtimes:
11:25 PASS landice_antarctica_mesh_gen
Total runtime 11:25
PASS: All passed successfully!
```

File: compass/landice/tests/antarctica/mesh.py [hunk 7]
```diff
@@ -187,8 +208,9 @@ def build_cell_width(self):
 
         # Calculate distance from each grid point to ice edge
         # and grounding line, for use in cell spacing functions.
-        distToEdge, distToGL = get_dist_to_edge_and_GL(self, thk, topg, x1,
-                                                       y1, window_size=1.e5)
+        distToEdge, distToGL = get_dist_to_edge_and_GL(self, thk, topg, x1, y1,
+                                                       section='antarctica',
```
[the remark this answers is not in the document]
Comment (hollyhan) [reply compass/landice/tests/antarctica/mesh.py:212]: @xylar, I asked @trhille about the appropriate values for `high_dist_bed` and `low_dist_bed` - apparently, these options were used/tested for the Greenland-mesh generation PR from which I copied to create this PR. Trevor's response was that the two values get used only when `use_bed = True`, and we can turn that option off for the moment as we need to do some testing to come up with the appropriate values for `high_dist_bed` and `low_dist_bed`. I think you can go ahead with merging, and we'll create a new PR when we get to testing of those values?

[the remark this answers is not in the document]
Comment (xylar) [reply compass/landice/tests/antarctica/mesh.py:212]: @hollyhan, it seems like your comment disappeared for me.

[the remark this answers is not in the document]
Comment (hollyhan) [reply compass/landice/tests/antarctica/mesh.py:212]: @xylar, the config file is update now, and the PR is ready to be merged.
I should note that `high_dist_bed` and `low_dist_bed` are only used when `use_bed = True`, and we should use the testgroup with `use_bed=False` until we further test/decide on appropriate values to use for them, which would be the task for the next PR.

**Comment by xylar** [comment]
The CI error is unrelated to this PR.

**Comment by xylar** [comment]
Thanks so much, @hollyhan!

**Pull request merged**